pkg/core/native/policy.go

```diff
@@ -0,0 +1,395 @@
+	ba, err := p.GetBlockedAccountsInternal(ic)
+	if err != nil {
+		panic(err)
+	}
+	return ba.ToStackItem()
+}
+
+// GetBlockedAccountsInternal returns list of blocked accounts hashes.
+func (p *Policy) GetBlockedAccountsInternal(ic *interop.Context) (BlockedAccounts, error) {
+	si := ic.DAO.GetStorageItem(p.Hash, blockedAccountsKey)
+	if si == nil {
+		return nil, errors.New("BlockedAccounts uninitialized")
+	}
+	ba, err := BlockedAccountsFromBytes(si.Value)
+	if err != nil {
+		return nil, err
+	}
+	return ba, nil
+}
+
+// setMaxTransactionsPerBlock is Policy contract method and  sets the upper limit
+// of transactions per block.
+func (p *Policy) setMaxTransactionsPerBlock(ic *interop.Context, args []stackitem.Item) stackitem.Item {
+	ok, err := p.checkValidators(ic)
+	if err != nil {
+		panic(err)
+	}
+	if !ok {
+		return stackitem.NewBool(false)
+	}
+	value := toBigInt(args[0])
+	err = p.setUint32WithKey(ic, maxTransactionsPerBlockKey, uint32(value.Int64()))
+	if err != nil {
+		panic(err)
+	}
+	return stackitem.NewBool(true)
+}
+
+// setMaxBlockSize is Policy contract method and sets maximum block size.
+func (p *Policy) setMaxBlockSize(ic *interop.Context, args []stackitem.Item) stackitem.Item {
+	ok, err := p.checkValidators(ic)
+	if err != nil {
+		panic(err)
+	}
+	if !ok {
+		return stackitem.NewBool(false)
+	}
+	value := toBigInt(args[0])
+	if payload.MaxSize <= value.Int64() {
+		return stackitem.NewBool(false)
+	}
+	err = p.setUint32WithKey(ic, maxBlockSizeKey, uint32(value.Int64()))
+	if err != nil {
+		panic(err)
+	}
+	return stackitem.NewBool(true)
+}
+
+// setFeePerByte is Policy contract method and sets transaction's fee per byte.
+func (p *Policy) setFeePerByte(ic *interop.Context, args []stackitem.Item) stackitem.Item {
+	ok, err := p.checkValidators(ic)
+	if err != nil {
+		panic(err)
+	}
+	if !ok {
+		return stackitem.NewBool(false)
+	}
+	value := toBigInt(args[0])
+	err = p.setInt64WithKey(ic, feePerByteKey, value.Int64())
+	if err != nil {
+		panic(err)
+	}
+	return stackitem.NewBool(true)
+}
+
+// blockAccount is Policy contract method and adds given account hash to the list
+// of blocked accounts.
+func (p *Policy) blockAccount(ic *interop.Context, args []stackitem.Item) stackitem.Item {
+	ok, err := p.checkValidators(ic)
+	if err != nil {
+		panic(err)
+	}
+	if !ok {
+		return stackitem.NewBool(false)
+	}
+	value := toUint160(args[0])
+	si := ic.DAO.GetStorageItem(p.Hash, blockedAccountsKey)
+	if si == nil {
+		panic("BlockedAccounts uninitialized")
+	}
+	ba, err := BlockedAccountsFromBytes(si.Value)
+	if err != nil {
+		panic(err)
+	}
+	indexToInsert := 0
+	if len(ba) != 0 {
+		indexToInsert = sort.Search(len(ba), func(i int) bool {
+			return !ba[i].Less(value)
+		})
+		if indexToInsert != len(ba) && ba[indexToInsert].Equals(value) {
+			return stackitem.NewBool(false)
+		}
+	}
+	ba = append(ba[0:indexToInsert], append([]util.Uint160{value}, ba[indexToInsert:]...)...)
```

Comment:
It's quite inefficient as you're creating a new slice in internal append. Take a look at b567ce86ac2261342ee4c5451c5a209c7590ddf1, it's the same problem essentially.

Reply:
Replaced this by `copy`.